lib/spack/spack/binary_distribution.py

```diff
@@ -1540,7 +1540,9 @@ def push_keys(*mirrors, **kwargs):
                 filename = fingerprint + '.pub'
 
                 export_target = os.path.join(prefix, filename)
-                spack.util.gpg.export_keys(export_target, fingerprint)
+
+                # Export public keys (private is set to False)
+                spack.util.gpg.export_keys(export_target, False, fingerprint)
```

Comment:
I get incredibly nervous seeing booleans passed as positional arguments. Could we consider providing it as a keyword argument instead? This may allow us to remove the comment.

I suppose I understand the desire to ensure the argument is always provided — defaulting `secret=None` in the `export_keys()` method, and checking if it’s `None`, is typically how I’ve seen this accomplished.

Reply:
In my change, I simply changed the `*keys` splat argument into a normal argument because unlike with `SpackCommand`, the keys are a genuine iterable of the same object.

Reply:
I was bothered by it too! The issue is that the function takes an unknown number of args (the names for the files). So maybe instead of that we would just require a list.